<?xml version="1.0"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
	<xacro:macro name="hitbot_transmission">
		<!-- Transmission 1 -->
		<transmission name="tran1">
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint1">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			</joint>
			<actuator name="motor1">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
				<mechanicalReduction>1</mechanicalReduction>
			</actuator>
		</transmission>
		<!-- Transmission 2 -->
		<transmission name="tran2">
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint2">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			</joint>
			<actuator name="motor2">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
				<mechanicalReduction>1</mechanicalReduction>
			</actuator>
		</transmission>
		<!-- Transmission 3 -->
		<transmission name="tran3">
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint3">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			</joint>
			<actuator name="motor3">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
				<mechanicalReduction>1</mechanicalReduction>
			</actuator>
		</transmission>
		<!-- Transmission 4 -->
		<transmission name="tran4">
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint4">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			</joint>
			<actuator name="motor4">
				<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
				<mechanicalReduction>1</mechanicalReduction>
			</actuator>
		</transmission>
	</xacro:macro>
</robot>